delivery/common/jwt/lib/jwt.js

```diff
@@ -0,0 +1,94 @@
+        }
+        return {
+            header: jwtHeader,
+            payload: jwtPayload
+        };
+    }
+    validateOptionTypes() {
+        if (void 0 !== this.jwtOptions.issuer && ("string" != typeof this.jwtOptions.issuer || 0 === this.jwtOptions.issuer.trim().length)) throw new Error("Invalid jwtOptions: issuer must be non empty string");
+        if (void 0 !== this.jwtOptions.audience && ("string" != typeof this.jwtOptions.audience || 0 === this.jwtOptions.audience.trim().length)) throw new Error("Invalid jwtOptions: audience must be non empty string");
+        if (void 0 !== this.jwtOptions.subject && ("string" != typeof this.jwtOptions.subject || 0 === this.jwtOptions.subject.trim().length)) throw new Error("Invalid jwtOptions: subject must be non empty string");
+        if (void 0 === this.jwtOptions.ignoreExpiration) this.jwtOptions.ignoreExpiration = !0; else if ("boolean" != typeof this.jwtOptions.ignoreExpiration) throw new Error("Invalid jwtOptions: ignoreExpiration must be boolean");
+        if (void 0 === this.jwtOptions.ignoreNotBefore) this.jwtOptions.ignoreNotBefore = !0; else if ("boolean" != typeof this.jwtOptions.ignoreNotBefore) throw new Error("Invalid jwtOptions: ignoreNotBefore must be boolean");
+        if (void 0 === this.jwtOptions.clockTolerance) this.jwtOptions.clockTolerance = 60; else if ("number" != typeof this.jwtOptions.clockTolerance) throw new Error("Invalid jwtOptions: clockTimestamp must be number");
+    }
+    async validate(base64JWTToken, alg, key) {
+        if ("string" != typeof base64JWTToken || "string" != typeof alg || "string" != typeof key) throw new Error("Invalid arguments!");
+        const jwtParts = base64JWTToken.split(".");
+        switch (alg) {
+          case "RS256":
+            try {
+                const cryptoKey = await crypto.subtle.importKey("spki", pem2ab(key), {
+                    name: "RSASSA-PKCS1-v1_5",
+                    hash: "SHA-256"
+                }, !1, [ "verify" ]);
+                return await crypto.subtle.verify({
+                    name: "RSASSA-PKCS1-v1_5"
+                }, cryptoKey, base64url.decode(jwtParts[2], "Uint8Array").buffer, (new TextEncoder).encode(`${jwtParts[0]}.${jwtParts[1]}`));
+            } catch (error) {
+                throw error;
+            }
+
+          case "HS256":
+            try {
+                const iKey = await crypto.subtle.importKey("raw", base16.decode(key, "Uint8Array").buffer, {
+                    name: "HMAC",
+                    hash: "SHA-256"
+                }, !1, [ "verify" ]);
+                return await crypto.subtle.verify({
+                    name: "HMAC"
+                }, iKey, base64url.decode(jwtParts[2], "Uint8Array").buffer, (new TextEncoder).encode(`${jwtParts[0]}.${jwtParts[1]}`));
+            } catch (error) {
+                throw error;
+            }
+
+          default:
+            throw new Error(`${alg} is not supported at the moment`);
```

Comment:
It would be handy to include the ECDSA and RSASSA-PSS algorithms, since they are available in our crypto libs.

Reply:
The additional algos are supported.